### ISSUE TITLE
Enhancement: User Specified JVM + JVM Arguments Support.

### ISSUE THREAD

**Issue by Aetopia** [Sol-Client/Client#65]
If possible consider adding support for the user to specify their JVM/JRE to use alongside custom JVM arguments.

**Comment by TheKodeToad** [comment]
Currently, I have created a JVM option that appends the chosen arguments. Would it be more useful to be able to change existing ones? I personally don't think so, since specifying arguments more than once usually overwrites the previous.

**Comment by Aetopia** [comment]
From my perspective, why one should be allowed to change the existing JRE/JVM Arguments.

1. Access to more optimizations.
2. Potential Performance Gain.
3. Using a better JRE, can provide better performance.

Another reason is that, it seems Sol Client performs poorly on my system with a i7-10700K + GTX 1650 + 24 GB.
OpenJDK + Sol Client's JVM Args seems to be the core reason why it is performing poorly.

**Comment by TheKodeToad** [comment]
I was also going to add an option to change the JRE. Can you explain what arguments that need to be changed/removed - since new additional flags usually take precedence?

This is the command that will be generated:
```
<user selected jvm folder>/bin/java
-Djava.library.path=<natives folder>
-Dio.github.solclient.client.version=<client version>
-Dio.github.solclient.client.secret=<launcher <-> client message secret>
-Xmx(memory)M
-Dio.github.solclient.client.discord_lib=<discord natives>
-Duser.language=en
-Duser.country=US
-Dfile.encoding=UTF-8
windows
    -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump

user arguments specified in settings
    varies
```

**Comment by Aetopia** [comment]
Actually None, infact i would recommend to have the `-Xmx` Arg accessible/editable by the user.

**Comment by Aetopia** [comment]
![image](https://user-images.githubusercontent.com/41850963/177370200-58b7e896-9f98-45fe-b049-1af7901c063f.png)
Pretty much just have the Editable JRE/JVM fields work like the MC Launcher.

Where it displays the default path/args to the user.

**Comment by TheKodeToad** [comment]
I don't think that will work since the default arguments may change from time to time.

**Comment by Aetopia** [comment]
> I don't think that will work since the default arguments may change from time to time. Instead of editing the existing -Xmx, the user can add -Xmx as an additional argument which will automatically remove the default.

Great idea! Maybe a warning for the ram allocation slider, incase the user decides to add the argument.

**Comment by TheKodeToad** [comment]
I have pushed the changes. Once released (or if you build and test it), this should be sufficient. If not, let me know what you are trying to do.
